include/bluetooth/conn_evt_prepare_cb.h

```diff
@@ -0,0 +1,66 @@
+/** @file
+ *  @brief Connection Event Prepare callback APIs
+ */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_CONN_PREPARE_CB_H__
+#define BT_CONN_PREPARE_CB_H__
+
+#include <stdint.h>
+#include <zephyr/bluetooth/conn.h>
+
+/**
+ * @file
+ * @defgroup bt_conn_evt_prepare_cb Connection Event Prepare callback
+ * @{
+ * @brief APIs to setup a connection event prepare callback
+ *
+ * The connection event prepare callbacks are triggered right before
+ * the start of a connection event. This allows the application to
+ * reduce the total system latency as data can be sampled right before
+ * it is sent on air.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Connection event prepare callback
+ *
+ * The callback is called from ISR context.
```

Comment:
would it be better to support thread mode instead so user can do anything in the callback, including BLE operations? As I see the primary goal is to call BLE API in timely manner. That also may be easier for us since thread mode by definition cannot be as accurate in terms of timing.

Reply:
@MarekPieta , do you have some input on this?

Reply:
Suggestion: Make this compile-time configurable through Kconfig where the default is to offload to thread mode

Reply:
may be we can start with thread mode only and add config later if needed? I guess applications that need accurate timing will use the conn event trigger directly anyway?